### ISSUE TITLE
Error: Unhandled HTTP response: 429 Too Many Requests

### ISSUE THREAD

**Issue by Kita2345** [TachibanaGeneralLaboratories/download-navi#211]
Error occurs after the download is complete. Error always occurs when downloading from MediaFire
![Screenshot_20220109-095754](https://user-images.githubusercontent.com/87817690/148667708-2c505ba2-b0b6-4b2b-a9c4-5d38acce8ebb.jpg)

**Comment by proninyaroslav** [comment]
Can you provide the link as example?

**Comment by Kita2345** [comment]
https://www.mediafire.com/file/8kvtd82hktizs5z/%255BBakadame.com%255D_Sewa_Pacar_Vol.15.rar/file

**Comment by proninyaroslav** [comment]
If I do it from the Navi browser everything works fine. Only thing: wrong filename, but I think it can be fixed.
![Screenshot_1641720550](https://user-images.githubusercontent.com/7840559/148676828-62d8b863-31c1-4f26-96df-ac228ec7806a.png) ![Screenshot_1641720573](https://user-images.githubusercontent.com/7840559/148676829-ddcf55f1-c532-4e4c-b8d1-14e01eb2216c.png)

**Comment by Kita2345** [comment]
the error occurs after the download is complete

**Comment by proninyaroslav** [comment]
Yes, the filename extracting was broken in the master branch with recent changes. I will fix it.

> the error occurs after the download is complete

Hmm, If I downloaded the file to the end and didn't notice any problems with **one** piece. Possibly that mediafire perceives multipart downloading as a DDoS attack.

**Comment by proninyaroslav** [comment]
Well, it seems mediafire no supports 16 concurrent connections. I've tried 1, 2, 15 pieces and it works fine. I think you should use fewer pieces for mediafire.
